packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -178,6 +196,142 @@ export class PaginatedMessage {
 		return this;
 	}
 
+	/**
+	 * Adds a page to the existing ones using a {@link MessageBuilder}. This will be added as the last page.
+	 * @param builder Either a callback whose first parameter is `new MessageBuilder()`, or an already constructed {@link MessageBuilder}
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 * const { MessageEmbed } = require('discord.js');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageBuilder((builder) => {
+	 * 		const embed = new MessageEmbed()
+	 * 			.setColor('#FF0000')
+	 * 			.setDescription('example description');
+	 *
+	 * 		return builder
+	 * 			.setContent('example content')
+	 * 			.setEmbed(embed);
+	 * });
+	 * ```
+	 * @example
+	 * ```typescript
+	 * const { MessageEmbed } = require('discord.js');
+	 * const { MessageBuilder, PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 *
+	 * const embed = new MessageEmbed()
+	 * 	.setColor('#FF0000')
+	 * 	.setDescription('example description');
+	 *
+	 * const builder = new MessageBuilder()
+	 * 	.setContent('example content')
+	 * 	.setEmbed(embed);
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addPageBuilder(builder);
+	 * ```
+	 */
+	public addPageBuilder(builder: MessageBuilder | ((builder: MessageBuilder) => MessageBuilder)): this {
+		return this.addPage(isFunction(builder) ? builder(new MessageBuilder()) : builder);
+	}
+
+	/**
+	 * Adds a page to the existing ones asynchronously using a {@link MessageBuilder}. This wil be added as the last page.
+	 * @param builder Either a callback whose first parameter is `new MessageBuilder()`, or an already constructed {@link MessageBuilder}
+	 * @example
+	 * ```typescript
+	 * const { PaginatedMessage } = require('@sapphire/discord.js-utilities');
+	 * const { MessageEmbed } = require('discord.js');
+	 *
+	 * const paginatedMessage = new PaginatedMessage()
+	 * 	.addAsyncPageBuilder(async (builder) => {
+	 * 		const someRemoteData = await fetch('https://contoso.com/api/users');
+	 *
+	 * 		const embed = new MessageEmbed()
+	 * 			.setColor('#FF0000')
+	 * 			.setDescription(someRemoteData.data);
+	 *
+	 * 		return builder
+	 * 			.setContent('example content')
+	 * 			.setEmbed(embed);
+	 * });
+	 * ```
+	 */
+	public addAsyncPageBuilder(builder: MessageBuilder | ((builder: MessageBuilder) => Promise<MessageBuilder>)): this {
+		return this.addPage(async () => (isFunction(builder) ? builder(new MessageBuilder()) : builder));
```

Comment:
This isn't using await at all.

Reply:
And why should it be separated? All pages are awaited, regardless of what they accept. In fact, this is redundant.